components/avatar/index.css

```diff
@@ -10,107 +10,123 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+  --spectrum-avatar-block-size: var(--spectrum-avatar-size-100)
 }
 
 .spectrum-Avatar--size200 {
-  @remapvars {
-    find: /--spectrum-avatar-size-200(.*)/;
-    filter: color;
-    replace: --spectrum-avatar$1;
-  }
+  --spectrum-avatar-inline-size: var(--spectrum-avatar-size-200);
+  --spectrum-avatar-block-size: var(--spectrum-avatar-size-200)
 }
 
 .spectrum-Avatar--size300 {
-  @remapvars {
-    find: /--spectrum-avatar-size-300(.*)/;
-    filter: color;
-    replace: --spectrum-avatar$1;
-  }
+  --spectrum-avatar-inline-size: var(--spectrum-avatar-size-300);
+  --spectrum-avatar-block-size: var(--spectrum-avatar-size-300)
 }
 
 .spectrum-Avatar--size400 {
-  @remapvars {
-    find: /--spectrum-avatar-size-400(.*)/;
-    filter: color;
-    replace: --spectrum-avatar$1;
-  }
+  --spectrum-avatar-inline-size: var(--spectrum-avatar-size-400);
+  --spectrum-avatar-block-size: var(--spectrum-avatar-size-400)
 }
 
 .spectrum-Avatar--size500 {
-  @remapvars {
-    find: /--spectrum-avatar-size-500(.*)/;
-    filter: color;
-    replace: --spectrum-avatar$1;
-  }
+  --spectrum-avatar-inline-size: var(--spectrum-avatar-size-500);
+  --spectrum-avatar-block-size: var(--spectrum-avatar-size-500)
 }
 
 .spectrum-Avatar--size600 {
-  @remapvars {
-    find: /--spectrum-avatar-size-600(.*)/;
-    filter: color;
-    replace: --spectrum-avatar$1;
-  }
+  --spectrum-avatar-inline-size: var(--spectrum-avatar-size-600);
+  --spectrum-avatar-block-size: var(--spectrum-avatar-size-600)
 }
 
 .spectrum-Avatar--size700 {
-  @remapvars {
-    find: /--spectrum-avatar-size-700(.*)/;
-    filter: color;
-    replace: --spectrum-avatar$1;
+  --spectrum-avatar-inline-size: var(--spectrum-avatar-size-700);
+  --spectrum-avatar-block-size: var(--spectrum-avatar-size-700)
+}
+
+@media (forced-colors: active) {
+  .spectrum-Avatar {
+    --highcontrast-avatar-color-opacity-disabled: 1;
+    --highcontrast-avatar-focus-indicator-color: CanvasText;
   }
 }
 
 .spectrum-Avatar {
   display: inline-block;
   position: relative;
-  inline-size: var(--spectrum-avatar-width);
-  block-size: var(--spectrum-avatar-height);
+  inline-size: var(--mod-avatar-inline-size, var(--spectrum-avatar-inline-size));
+  block-size: var(--mod-avatar-block-size, var(--spectrum-avatar-block-size));
 
-  border-radius: var(--spectrum-avatar-border-radius);
+  border-radius: var(--mod-avatar-border-radius, var(--spectrum-avatar-border-radius));
   border-width: 0;
 
+  outline: none;
+
   -webkit-user-drag: none;
   user-select: none;
-  overflow: hidden;
 
-  &:after {
-    content: '';
+  overflow: visible;
+
+  opacity: var(--mod-avatar-color-opacity, var(--spectrum-avatar-color-opacity));
+
+  &.is-disabled {
+    opacity: var(--highcontrast-avatar-color-opacity-disabled, var(--mod-avatar-color-opacity-disabled, var(--spectrum-avatar-color-opacity-disabled)));
+  }
+}
+
+.spectrum-Avatar:focus-within:not(.is-disabled) .spectrum-Avatar-link {
```

Comment:
This could also be: `.spectrum-Avatar:not(.is-disabled) .spectrum-Avatar-link:focus-visible` and have the same functional styles, right? The main difference that we've not discussed for `:focus-within` is that is doesn't match the keyboard heuristics of `:focus-visible`...

Reply:
Yeah, that'll work.

And yes, we definitely have a discussion about how best to serve `:focus-*` to y'all going forward.